### ISSUE TITLE
Workflows - Not picking the right Task List and Workflow History List

### ISSUE THREAD

**Issue by BigBrahma** [SharePoint/PnP-Sites-Core#819]
PnP engine successfully created the Workflows. But the task list is associated to the new;y created task list instead of default Tasks List. Same for Workflow History List.  I have used the latest version(Version=2.8.1610.1 and PnP/2016/05/ProvisioningSchema)

![image](https://cloud.githubusercontent.com/assets/18693266/19641706/8a518d10-99ff-11e6-8a6f-f8cbafbd817f.png)

![image](https://cloud.githubusercontent.com/assets/18693266/19641842/22739a16-9a00-11e6-9c9c-1f6c07ba2548.png)

Checked the TaskID and History ID. It is proper. See the below screenshots.
![image](https://cloud.githubusercontent.com/assets/18693266/19641925/8c2d16ee-9a00-11e6-9d1a-ca10040e55cb.png)

![image](https://cloud.githubusercontent.com/assets/18693266/19641978/c6db3834-9a00-11e6-9630-a9ba236c6627.png)

Tasks List and Workflow History list are already there in the target site.

Please let me know did i miss something?

**Comment by jansenbe** [comment]
Hi @BigBrahma,

If you want to provision your WF with their own dedicated task and history lists then you'll need to create these lists as part of your PnP template and link them the WF. This would mean giving them titles that differ from the names of the default task/history lists. 

Can you let us know if that solves your problem?

**Comment by jansenbe** [comment]
Closing this issue since we've not received feedback for a long time. We want to further investigate this, but if there's no feedback we can't. Absolutely fine for you to re-open the issue if you can still re-produce using the March 2017 release or if you have additional context that can help.

**Comment by Pieter-Veenstra** [comment]
@jansenbe , I just came across the same problem.

The way to reproduce is:

1. Create two team sites ( in my case 2 subsites within the same site collection)
2. Create a Workflow in SPD in sub site 1
3. From site 1 extract the site template using Get-PnPProvisioningTemplate
4. Apply the generated template to site 2. 

You now have a site with the template that has created the workflow on the list however it cannot find the workflow lists ( the lists do exist in the site!)

For what it's worth I'm using SP 2016

**Comment by Pieter-Veenstra** [comment]
I just tried the same again but I updated the tokens in the template.xml with the guids of the lists and now the task list and the history list are applied to the workflow. It looks like the tokens aren't resolved properly. 

I then created a 3 test site. Applied my template while my templates uses tokens like this again:

Key="HistoryListId" Value="{listid:TestWorkFlow History}"

then I deleted the workflow and applied my template again and now the lists are applied to the workflow as expected.

It looks like the token values are collected before the lists are created.

A workaround could be to run the Lists handler first and then apply the whole template but that isn't really a good workaround. 

I don't have a visual studio environment available where I can currently test the above thoughts by stepping through the code. But it looks like something is going wrong here.
